programming-extensions/programming-extension-vfsprovider/src/main/java/org/objectweb/proactive/extensions/vfsprovider/client/ProActiveFileObject.java

```diff
@@ -239,6 +261,47 @@ protected long doGetLastModifiedTime() throws Exception {
     }
 
     @Override
+    public List<FileObject> listFiles(final FileSelector selector) throws FileSystemException {
+        if (!exists() || selector == null) {
+            return null;
+        }
+
+        final ArrayList<FileObject> list = new ArrayList<FileObject>();
+        this.findFiles(selector, true, list);
+        return list;
+    }
+
+    @Override
+    public FileObject[] findFiles(final FileSelector selector) throws FileSystemException {
+        final List<FileObject> list = this.listFiles(selector);
+        return list == null ? null : list.toArray(new FileObject[list.size()]);
+    }
+
+    @Override
+    public void findFiles(final FileSelector selector, final boolean depthwise, final List<FileObject> selected)
+            throws FileSystemException {
+        if (forkJoinPool == null) {
```

Comment:
shouldn't we protect forkJoinPool accesses as it is static and shared between all ProActiveFileObject?